omnigibson/utils/usd_utils.py

```diff
@@ -689,6 +691,63 @@ def mesh_prim_to_trimesh_mesh(mesh_prim, include_normals=True, include_texcoord=
 
     return trimesh.Trimesh(**kwargs)
 
+def mesh_prim_shape_to_trimesh_mesh(mesh_prim):
+    """
+    Generates trimesh mesh from @mesh_prim if mesh_type is "Sphere", "Cube", "Cone" or "Cylinder"
+
+    Args:
+        mesh_prim (Usd.Prim): Mesh prim to convert into trimesh mesh
+
+    Returns:
+        trimesh.Trimesh: Generated trimesh mesh
+    """
+    mesh_type = mesh_prim.GetPrimTypeInfo().GetTypeName()
+    if mesh_type == "Sphere":
+        radius = mesh_prim.GetAttribute("radius").Get()
+        trimesh_mesh = trimesh.creation.icosphere(subdivision=3, radius=radius)
+    elif mesh_type == "Cube":
+        extent = mesh_prim.GetAttribute("size").Get()
+        trimesh_mesh = trimesh.creation.box([extent] * 3)
+    elif mesh_type == "Cone":
+        radius = mesh_prim.GetAttribute("radius").Get()
+        height = mesh_prim.GetAttribute("height").Get()
+        trimesh_mesh = trimesh.creation.cone(radius=radius, height=height)
+        # Trimesh cones are centered at the base. We'll move them down by half the height.
+        transform = trimesh.transformations.translation_matrix([0, 0, -height / 2])
+        trimesh_mesh.apply_transform(transform)
+    elif mesh_type == "Cylinder":
+        radius = mesh_prim.GetAttribute("radius").Get()
+        height = mesh_prim.GetAttribute("height").Get()
+        trimesh_mesh = trimesh.creation.cylinder(radius=radius, height=height)
+    else:
+        raise ValueError(f"Expected mesh prim to have type Sphere, Cube, Cone or Cylinder, got {mesh_type}")
```

Comment:
It's really clean to centralize this logic, but it's a significant slowdown to compute these throwaway meshes compared to the previous approach which analytically computed the volumes etc. - I guess we're OK with that?

Reply:
I think this should be okay. Our code only compute the volume once when initializing the objects.